tests/integration/scie/test_pex_scie.py

```diff
@@ -737,6 +753,151 @@ def test_scie_busybox_console_script_injections(
     )
 
 
+@skip_if_no_provider
+@execution_mode_args
+def test_scie_busybox_console_script_inject_env(
+    tmpdir,  # type: Any
+    foo,  # type: str
+    bar,  # type: str
+    execution_mode_args,  # type: List[str]
+):
+    # type: (...) -> None
+
+    busybox = os.path.join(str(tmpdir), "busybox")
+    run_pex_command(
+        args=[
+            foo,
+            bar,
+            "-m",
+            "foo:one",
+            "--inject-env",
+            "FOO=bar",
+            "--scie",
+            "lazy",
+            "--scie-busybox",
+            "@bar,foo-script1",
+            "-o",
+            busybox,
+        ]
+        + execution_mode_args,
+        quiet=True,
+    ).assert_success()
+
+    assert b"FOO=bar foo1: injected?\n" == subprocess.check_output(
+        args=[busybox, "foo-script1", "injected?"]
+    )
+    assert b"FOO=baz foo1: injected?\n" == subprocess.check_output(
+        args=[busybox, "foo-script1", "injected?"], env=make_env(FOO="baz")
+    )
+    assert b"bar1: injected?\n" == subprocess.check_output(
+        args=[busybox, "bar-script1", "injected?"]
+    )
+    assert b"bar2: injected?\n" == subprocess.check_output(
+        args=[busybox, "bar-script2", "injected?"]
+    )
+
+    bin_dir = os.path.join(str(tmpdir), "bin_dir")
+    subprocess.check_call(args=[busybox, bin_dir], env=make_env(SCIE="install"))
+    assert sorted(["foo-script1", "bar-script1", "bar-script2"]) == sorted(os.listdir(bin_dir))
+
+    assert b"FOO=bar foo1: injected?\n" == subprocess.check_output(
+        args=[os.path.join(bin_dir, "foo-script1"), "injected?"]
+    )
+    assert b"FOO=baz foo1: injected?\n" == subprocess.check_output(
+        args=[os.path.join(bin_dir, "foo-script1"), "injected?"], env=make_env(FOO="baz")
+    )
+    assert b"bar1: injected?\n" == subprocess.check_output(
+        args=[os.path.join(bin_dir, "bar-script1"), "injected?"]
+    )
+    assert b"bar2: injected?\n" == subprocess.check_output(
+        args=[os.path.join(bin_dir, "bar-script2"), "injected?"]
+    )
+
+
+@skip_if_no_provider
+@execution_mode_args
+def test_scie_busybox_console_script_inject_python_args(
```

Comment:
The `--inject-python-args` needed no fixing, but I added the this test to make sure all 3 injection types were covered.